//------------------------------------------------------------------------------
// {C} Copyright 2019 Pensando Systems Inc. All rights reserved
// protobuf specification for control plane test configurations, 
// Used only for testing, not exposed to outside
//------------------------------------------------------------------------------

syntax = "proto3";
package pds_ms;

import "pdsa.proto";
import "meta/meta.proto";
import "types.proto";

// service definition
service CPTestSvc
{
  rpc CPL2fTestCreate (CPL2fTestCreateSpec) returns (CPL2fTestResponse) {}
  rpc CPL2fTestDelete (CPL2fTestDeleteSpec) returns (CPL2fTestResponse) {}
  rpc CPDestIPTrackTestCreate (CPDestIPTrackTestCreateSpec) returns (CPDestIPTrackTestResponse) {}
  rpc CPDestIPTrackTestDelete (CPDestIPTrackTestDeleteSpec) returns (CPDestIPTrackTestResponse) {}
}

message CPL2fTestCreateSpec {
  option (venice.pdsaSetGlobOpts) = {
    ActionFn: "l2f_test_local_mac_ip_add"
  };
    bytes           SubnetId = 1;
    types.IPAddress IPAddr   = 2;
    bytes           MACAddr  = 3;
    uint32          IfId     = 4;
}

message CPL2fTestDeleteSpec {
  option (venice.pdsaSetGlobOpts) = {
    ActionFn: "l2f_test_local_mac_ip_del"
  };
    bytes           SubnetId = 1;
    types.IPAddress IPAddr   = 2;
    bytes           MACAddr  = 3;
}

message CPL2fTestResponse {
    types.ApiStatus ApiStatus = 1;
}


message CPDestIPTrackTestCreateSpec {
  option (venice.pdsaSetGlobOpts) = {
    ActionFn: "destip_track_start"
  };
    types.IPAddress DestIP   = 1;
    uint32          PDSObjId    = 2;
}

message CPDestIPTrackTestDeleteSpec {
  option (venice.pdsaSetGlobOpts) = {
    ActionFn: "destip_track_stop"
  };
    types.IPAddress DestIP   = 1;
}

message CPDestIPTrackTestResponse {
    types.ApiStatus ApiStatus = 1;
}
